src/org/parosproxy/paros/core/scanner/VariantMultipartFormParameters.java

```diff
@@ -91,7 +92,11 @@ public void setMessage(HttpMessage msg) {
 				String value = part.replaceAll(boundary + partHeaderLine + HttpHeader.CRLF + HttpHeader.CRLF, "");
 				value = value.replaceAll(HttpHeader.CRLF + "(" + Pattern.quote(getBoundary(contentType)) + "--"
 						+ HttpHeader.CRLF + ")?$", ""); // Strip final boundary
-				extractedParameters.add(new NameValuePair(NameValuePair.TYPE_POST_DATA, name, value, position));
+				if (isFileParam) {
+					extractedParameters.add(new NameValuePair(NameValuePair.TYPE_MULTIPART_DATA_FILE_PARAM, name, value, position));
+				} else {
+					extractedParameters.add(new NameValuePair(NameValuePair.TYPE_POST_DATA, name, value, position));

```

Comment:
Should be `TYPE_MULTIPART_DATA_PARAM`? We should assert the types in `VariantMultipartFormParametersUnitTest.shouldExtractParametersFromAllParts`.
In `shouldNotAllowToModifyReturnedParametersList` it could be changed to use one of the multipart constants (just for consistency).